cmd/snap-update-ns/utils.go

```diff
@@ -45,6 +47,87 @@ var (
 	sysFchown  = syscall.Fchown
 )
 
+// Create directories for all but the last segments and return the file
+// descriptor to the leaf directory. This function is a base for secure
+// variants of mkdir, touch and symlink.
+func secureMkPrefix(segments []string, perm os.FileMode, uid, gid int) (int, error) {
+	logger.Debugf("secure-mk-prefix %q %v %d %d -> ...", segments, perm, uid, gid)
+
+	// Declare var and don't assign-declare below to ensure we don't swallow
+	// any errors by mistake.
+	var err error
+	var fd int
```

Comment:
I'd init fd to -1 to be on the safe side.

Reply:
Commented in the other PR that we cannot do that as it triggers ineffectual assignment error.